### DISCUSSION TITLE
Qt 6 Migration

### DISCUSSION THREAD

**Discussion by coder3101** [cpeditor/cpeditor#707] [Development]
Qt 6.0 was released recently with new features and more cleaner internal implementation of various Qt classes. This update comes under a new package ([qt6-base](https://www.archlinux.org/packages/extra/x86_64/qt6-base/)). The complete release blog that summerizes most of the new feature and changes can be found [here](https://www.qt.io/blog/qt-6.0-released)

I have tried compiling our codebase against the Qt6 and it turns out it won't be much harder to update. A detailed [migration guide](https://doc.qt.io/qt-6/portingguide.html) by Qt can help us smoothly transition to the new version. Most of the breaking changes are for Qt Graphics API which we do not use. All Deprecated modules and classes in Qt5.15 have been removed from Qt6, one such class is `QRegEx`, widely used in cpeditor and its dependencies.

The migration plan should be as follows:
- [ ] Enable C++17 in cpeditor and its dependencies
- [ ] Replace `Qt5` in `CMakeLists.txt` with `Qt6` in cpeditor and dependencies. singleapplication already [has Qt6 support](https://github.com/itay-grudev/SingleApplication/blob/aede311d28d20179216c5419b581087be2a8409f/CMakeLists.txt#L13)
- [ ] Refactor code so that it can compile with Qt6. 
- [ ] After we can locally compile with Qt 6, we need to start modifying Github Action scripts so CI can build with Qt6
- [ ] Check the release actions (like AUR, DEB) are being able to build with Qt6
- [ ] Change documentation on website and Readme wherever it mentions Qt5 to Qt6

**Comment by ouuan** [comment]
If there's no specific reason, I suggest waiting until Qt 6.1 when the API are more stable and more bugs are fixed.

**Reply by coder3101** [reply]
Sure thing!
